From 5bf365fa3db259e1dc9c475f29c0a0374b189c17 Mon Sep 17 00:00:00 2001
From: Shayne Chen <shayne.chen@mediatek.com>
Date: Tue, 24 Jun 2025 11:20:44 +0800
Subject: [PATCH 211/236] mtk: hostapd: add 6g check before
 hostapd_config_check_bss_6g()

Otherwise the following log will keep printing on non-6g band:
<27>Jun 20 22:31:09 OpenWrt hostapd: Management frame protection is required in 6 GHz

Signed-off-by: Shayne Chen <shayne.chen@mediatek.com>
---
 src/ap/ap_drv_ops.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/ap/ap_drv_ops.c b/src/ap/ap_drv_ops.c
index 21712f489..19bd9d91c 100644
--- a/src/ap/ap_drv_ops.c
+++ b/src/ap/ap_drv_ops.c
@@ -1223,7 +1223,8 @@ void hostapd_get_hw_mode_any_channels(struct hostapd_data *hapd,
 {
 	int i;
 	bool is_no_ir = false;
-	bool allow_6g_acs = hostapd_config_check_bss_6g(hapd->conf) &&
+	bool allow_6g_acs = is_6ghz_op_class(hapd->iconf->op_class) &&
+		hostapd_config_check_bss_6g(hapd->conf) &&
 		(hapd->iface->conf->ieee80211ax ||
 		 hapd->iface->conf->ieee80211be);
 
-- 
2.45.2

